packages/runtime/src/widgets/Search.tsx

```diff
@@ -61,7 +63,7 @@ export const Search: React.FC<SearchProps> = ({
   });
 
   const { id, rootRef, values } = useRegisterBindings(
-    { styles, value, ...rest, widgetName },
+    { styles, value, ...rest, widgetName, initialValue },
```

Comment:
Subtle but important distinction - do you also want the `value` in widget state to be initialized to initialValue? This will update the search results, allow it to be referenced via variable, etc. Basically, is this only a visual indicator?

If so, you will need to add a hook to update the local component `value` state variable.

If not, then you should make this a separate prop called `initialValue` to avoid confusion.

Reply:
@evshi i'm intending on using it like the default `value` of the `Dropdown`

Reply:
@justEhmadSaeed then you need to add the useEffect hook that syncs the initialValue to the widget state value.

Reply:
@evshi updated the PR to sync the widget state value with that of the initialValue